tooling/minimize/src/enums.rs

```diff
@@ -0,0 +1,91 @@
+use crate::*;
+
+pub fn translate_enum<'tcx>(
+    ty: rs::Ty<'tcx>,
+    adt_def: rs::AdtDef<'tcx>,
+    sref: rs::GenericArgsRef<'tcx>,
+    tcx: rs::TyCtxt<'tcx>,
+) -> Type {
+    let a = rs::ParamEnv::empty().and(ty);
+    let layout = tcx.layout_of(a).unwrap().layout;
+    let size = translate_size(layout.size());
+    let align = translate_align(layout.align().abi);
+
+    // The discriminant type in Rust is always isize which always gets translated as i64.
+    let discriminant_ty = IntType { signed: Signedness::Signed, size: Size::from_bytes_const(8) };
+
+    let (variants, discriminator) = match layout.variants() {
+        rs::Variants::Single { index } => {
+            let fields = translate_fields(layout.fields(), adt_def.variant(*index), sref, tcx);
+            ([Variant { ty: Type::Tuple { fields, size, align }, tagger: Map::new() }].into_iter().collect(), Discriminator::Known(0.into()))
+        },
+        rs::Variants::Multiple {
+            tag,
+            tag_encoding,
```

Comment:
I think you'll want to do `tag_encoding: rs::TagEncoding::Direct` here; the niche one will need quite a bit of different logic probably.